Automattic/wp-calypso
# Checkout: Move checkout line items into CSS Grid containers

This PR rearranges the checkout sidebar line items into CSS Grid containers to provide a more flexible layout.

This PR also refactors existing styled components and adds a few more in key locations.

Related to https://github.com/Automattic/payments-shilling/issues/1969

## Proposed Changes

* A number of changes to the `LineItem` et al styled components found in `packages/wpcom-checkout/src/checkout-line-items.tsx` to wrap them in a CSS Grid

## Testing Instructions

* Add a couple of different products to your cart, i.e plans, domains, email, etc
* Go to `http://calypso.localhost:3000/checkout/your-site-here?checkoutVersion=2` note the query parameter at the end
* Check that the items in the sidebar are spaced evenly
* Check on smaller device widths
* Try to find a product or use case that breaks the layout

Example products and use cases to test:
- [ ] Dotcom Plans
- [ ]  Jetpack Products
- [ ]  Domain registrations
- [ ]  Domain transfers
- [ ]  Domain connections
- [ ]  Woo Express plans
- [ ]  Email licenses `'%(numberOfMailboxes)d mailbox for %(domainName)s'`
- [ ]  DIFM contracts `'%(numberOfExtraPages)d Extra Page: %(costOfExtraPages)s one-time fee'`
- [ ]  P2Plus `'Monthly subscription: %(itemPrice)s x %(members)s member'`
- [ ]  Tiered volume space addons `'%(quantity)s GB extra space, %(price)s per year'`
- [ ]  Jetpack search `'Up to %(purchaseQuantityDividedByThousand)sk records and/or requests per month'`
- [ ]  Partner logos (not sure when these are shown but it states `Included in your IONOS plan`)
- [ ]  Gift purchases
- [ ]  Large prices
- [ ]  Different currencies
- [ ]  Different languages (big 16?)
- [ ]  What are partner coupons and how are they displayed
